Source/AssetRipper.IO.Files/Streams/Smart/SmartStream.cs

```diff
@@ -95,6 +95,16 @@ public SmartStream CreateReference()
 		return new SmartStream(this);
 	}
 
+	[MemberNotNull(nameof(Stream))]
+	public SmartStream CreateLogicalFile(long offset, long size)
+	{
+		ThrowIfNull();
+		return new SmartStream(this)
+		{
+			Stream = new RandomAccessStream(Stream, offset, size),
+		};
```

Comment:
This will need changes. It should check the underlying stream type and if compatible, return a new SmartStream using the same ref counter. If not compatible, the segment should be copied.